apps/backend/src/libs/plugins/auth-plugin/auth-plugin.ts

```diff
@@ -0,0 +1,38 @@
+import { type FastifyPluginCallback } from "fastify";
+
+import { HTTPCode } from "~/libs/modules/http/libs/enums/enums.js";
+import { ServerHooks } from "~/libs/plugins/libs/enums/enums.js";
+
+type PluginOptions = {
+	excludedRoutePrefixes?: string[];
+	tokenVerifier: (token: string) => Promise<unknown>;
```

Comment:
```suggestion
	whiteRoutes?: string[];
	token: Token
```
Should send full token service here
Add user service here as well

Reply:
Do I need to pick commits from PR #40 or create a `Token` abstraction based on that PR?

Reply:
You can do that, yes